DEX.md

```diff
@@ -0,0 +1,93 @@
+
+# Market Ratio
+
+A liquidity pool maintains a market ratio (MR) between its two tokens:
+
+$MR = \frac{reserve_0}{reserve_1}$
+
+Equivalently, it is the price of $T_1$ quoted in $T_0$. Intuitively, the higher the relative abundance of $T_0$ compared to $T_1$, the cheaper $T_1$ becomes in terms of $T_0$.
+
+When an LP deposits $amount_0$ and $amount_1$ to the pool (deposit ratio $DR = \frac{amount_0}{amount_1}$), pool takes from $amount_0$ and $amount_1$ according to the market ratio:
+
+1. if $DR == MR$ => pool takes $amount_0$ and $amount_1$
+2. if $DR >  MR$ => pool takes $amount_1 \times MR$ from $amount_0$ and $amount_0$
+3. if $DR <  MR$ => pool takes $amount_0$ and $amount_0 \times MR$ from $amount_1$
+
+This ensures that liquidity is always added proportional to the current reserves' balances.
+If $DR \ne MR$, the excess amount is returned to the LP.
+
+# Trading
+
+Assume a trader wants to swap $T_0$ for $T_1$. To trade, they send $amount_0$ of $T_0$.
+
+In the uniform liquidity model (used in Uniswap V2), liquidity is defined as:
+
+$L^2 = reserve_0 * reserve_1$
+
+This formula is known as *Constant Product Invariant*. When trading, the liquidity remains unchanged:
+
+$reserve_0 * reserve_1 = (reserve_0 + amount_0) * (reserve_1 - amount_1)$
+
+$amount_1$ is the amount of $T_1$ that is going to be removed from the pool. Before sending
+it to the user, some fees are deducted: mainly a platform fee that goes to the owners of the
+DEX and an LP-fee that is going to be accumulated for the liquidity providers proportional to the 
+amount of liquidity they have added to the system.
+
+# Withdrawing Liquidity
+
+Liquidity providers can withdraw their tokens at any time. However, due to trades happening in between, they rarely receive the same token amounts they initially deposited. Instead, they withdraw, depending on the net direction of trades:
+
+1. More of one token,
+2. Less of the other, and
+3. The fees they have accrued during trading activity.
+
+Some DEXes provide the LPs with the functionality to withdraw their accrued fee and, if desired, reinvest it to
+the protocol to compound their rewards.
+
+# Pitfalls and Fallacies
+
+In this section, we will highlight common pitfalls and misconceptions that can be confusing for users.
+
+## Pool Price $\ne$ Actual Price
+
+The pool exchange price (which is called *Local Price*) is solely defined by the ratio of assets in the pool. As users can trade their tokens, the ratio of the tokens diverges from the actual price.
+
+## Impermanent Loss
+
+When LPs deposit assets to a pool, the exchange price can change according to the point $MR$. As mentioned above, the local price can be different from the actual price. Hence, if one asset, e.g., $T_0$, is significantly more appreciated than the other one. In this case, liquidity providers end up selling their assets at an undervalued price relative to the broader market. 
```

Comment:
```suggestion
When LPs deposit assets to a pool, the exchange price can change according to the point $MR$. As mentioned above, the local price can be different from the actual price. Hence, one asset, e.g., $T_0$, is significantly more appreciated than the other one. In this case, liquidity providers end up selling their assets at an undervalued price relative to the broader market. 
```
(something just seemed to be a bit off - please check if this is the right meaning)